spec.html

```diff
@@ -7573,134 +7573,44 @@ <h1>GetGlobalObject ( )</h1>
-        </tbody>
-      </table>
-    </emu-table>
-    <p>A <dfn id="job-queue">Job Queue</dfn> is a FIFO queue of PendingJob records. Each Job Queue has a name and the full set of available Job Queues are defined by an ECMAScript implementation. Every ECMAScript implementation has at least the Job Queues defined in <emu-xref href="#table-26"></emu-xref>.</p>
-    <p>Each agent has its own set of named Job Queues.  All references to a named job queue in this specification denote the named job queue of the surrounding agent.</p>
-    <emu-table id="table-26" caption="Required Job Queues">
-      <table>
-        <tbody>
-        <tr>
-          <th>
-            Name
-          </th>
-          <th>
-            Purpose
-          </th>
-        </tr>
-        <tr>
-          <td>
-            ScriptJobs
-          </td>
-          <td>
-            Jobs that validate and evaluate ECMAScript |Script| and |Module| source text. See clauses 10 and 15.
-          </td>
-        </tr>
-        <tr>
-          <td>
-            PromiseJobs
-          </td>
-          <td>
-            Jobs that are responses to the settlement of a Promise (see <emu-xref href="#sec-promise-objects"></emu-xref>).
-          </td>
-        </tr>
-        </tbody>
-      </table>
-    </emu-table>
-    <p>A request for the future execution of a Job is made by enqueueing, on a Job Queue, a PendingJob record that includes a Job abstract operation name and any necessary argument values. When there is no running execution context and the execution context stack is empty, the ECMAScript implementation removes the first PendingJob from a Job Queue and uses the information contained in it to create an execution context and starts execution of the associated Job abstract operation.</p>
-    <p>The PendingJob records from a single Job Queue are always initiated in FIFO order. This specification does not define the order in which multiple Job Queues are serviced. An ECMAScript implementation may interweave the FIFO evaluation of the PendingJob records of a Job Queue with the evaluation of the PendingJob records of one or more other Job Queues. An implementation must define what occurs when there are no running execution context and all Job Queues are empty.</p>
+  <emu-clause id="sec-jobs">
+    <h1>Jobs and Host Operations to Enqueue Jobs</h1>
+    <p>A <dfn id="job">Job</dfn> is an abstract closure with no parameters that initiates an ECMAScript computation when no other ECMAScript computation is currently in progress.</p>
+    <p>Jobs are scheduled for execution by ECMAScript host environments with abstract operations. The ECMAScript specification describes the host hook HostEnqueuePromiseJob to schedule one job; host environments may define additional abstract operations which schedule jobs. Such operations accept a Job abstract closure as the parameter and schedule it to be performed at some future time. Their implementations must conform to the following requirements:</p>
```

Comment:
> host environments with abstract operations

"with abstract operations" reads kinda weird. Did you mean "with host hooks"?

Reply:
I think the original intention was maybe like "with abstract operations that ultimately call host hooks"? I do agree "with host hooks" sounds better though, will change.

Reply:
Also the

> describes the host hook HostEnqueuePromiseJob to schedule one job

part is kind of weird. Maybe it would be better as "to schedule one **kind of** job"? Alternatively something like

> The ECMAScript specification describes the host hook HostEnqueuePromiseJob as one way to schedule jobs; host environments may define additional abstract operations which schedule jobs.